### DISCUSSION TITLE
Confusion around what `getRef` inside `createStyles` is used for

### DISCUSSION THREAD

**Discussion by shutallbridge** [mantinedev/mantine#1277] [Q&A] [answered]
I'm confused about what `getRef` does and what it's used for. I have a couple of questions after reading [this section](https://mantine.dev/theming/create-styles/#composition-and-nested-selectors) of the documentation.

- I can't seem to find `ref` as any CSS property. What is the purpose of `ref: getRef()` and what does it do? Why is it needed inside `.button`?
- What does the documentation mean by "static selector"? And when is it useful to be used/created with `getRef`?
- Is the `&` selecting the `container` class similar to how Emotion works? And how is this different from `getRef`?

```tsx
// some lines omitted for brevity

const useStyles = createStyles((theme, _params, getRef) => ({
  button: {
    // assign reference to selector
    ref: getRef('button'),

    // and add any other properties
    backgroundColor: theme.colors.blue[6],
    color: theme.white,
  },

  container: {
    // reference button with nested selector
    [`&:hover .${getRef('button')}`]: {
      backgroundColor: theme.colors.violet[6],
    },
  },
}));
```

**Comment by rtivital** [comment, marked as answer]
`getRef` is used to assign a static class to an element that can be used to create selectors that involve this class,
for example, for the following markup and css:

```html
<button class="button">
   <span class="button-label">Button label</span>
</button>

<style>
.button > .button-label {
  color: red;
}
</style>
```

createStyles equivivalent will look like this:

```tsx
createStyles((theme, params, getRef) => ({
  label: { ref: getRef('label') },
  button: {
    [`& .${getRef('label')}`]: { color: 'red' }
  },
}))
```

**Reply by shutallbridge** [reply]
Got it! Thanks for clarifying my confusion!

**Reply by milesingrams** [reply]
Could this be done by just defaulting to the property key?
I feel like 99% of the time you would just be doing:

```
  myKey: {
    ref: getRef('myKey'),
  }
```

Why not just make every key automatically generate a similarly named ref unless a different one is specified?
